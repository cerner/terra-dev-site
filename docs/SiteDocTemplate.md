# SiteDocTemplate
Provides a standard and easily adjustable layout for all documentation pages.

## Getting Started

- Install with [npmjs](https://www.npmjs.com):
  - `npm install terra-dev-site`
  - `yarn add terra-dev-site`

## Usage
```
import { version } from '../package.json';
import readme from './README.md';

<SiteDocTemplate version={version} readme={readme} />
```
See also: the example below, demonstrating a more complete SiteDocTemplate configuration