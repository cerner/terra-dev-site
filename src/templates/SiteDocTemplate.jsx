import React from 'react';

const propTypes = {
};

const defaultProps = {
};

class SiteDocTemplate extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return null;
  }
}

SiteDocTemplate.propTypes = propTypes;
SiteDocTemplate.defaultProps = defaultProps;

export default SiteDocTemplate;
