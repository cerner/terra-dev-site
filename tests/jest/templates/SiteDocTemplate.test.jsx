import React from 'react';

import SiteDocTemplate from '../../../src/templates/SiteDocTemplate';
import readme from '../../../examples/test-examples/templates/TestReadme.md';
// eslint-disable-next-line import/no-duplicates
import TestComponentExample from '../../../examples/test-examples/templates/TestComponentExample';
// Had to remove !rawloader! from the next two imports as it broke jest, but in normal use it should be there
// eslint-disable-next-line import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions, import/no-duplicates
import exampleSrc from '../../../examples/test-examples/templates/TestComponentExample.jsx';
// eslint-disable-next-line import/no-webpack-loader-syntax, import/first, import/no-unresolved, import/extensions
import testComponentSrc from '../../../examples/test-examples/templates/TestComponent.jsx';

jest.mock('../../../examples/test-examples/templates/TestReadme.md', () => 'test-file-stub');
jest.mock('../../../examples/test-examples/templates/TestComponentExample.jsx', () => 'test-file-stub');
jest.mock('../../../examples/test-examples/templates/TestComponent.jsx', () => 'test-file-stub');
const exampleElement = <TestComponentExample />;
// Snapshot Tests
it('should render a default component with nothing', () => {
  const wrapper = shallow(<SiteDocTemplate />);
  expect(wrapper).toMatchSnapshot();
});

it('should show the version', () => {
  const wrapper = shallow(<SiteDocTemplate version="1.9" />);
  expect(wrapper).toMatchSnapshot();
});

it('should show the readme', () => {
  const wrapper = shallow(<SiteDocTemplate readme={readme} />);
  expect(wrapper).toMatchSnapshot();
});

it('should show the version and readme', () => {
  const wrapper = shallow(<SiteDocTemplate version="1.9" readme={readme} />);
  expect(wrapper).toMatchSnapshot();
});

it('should show one example', () => {
  const wrapper = shallow(<SiteDocTemplate examples={[{ title: 'Test Example 1', description: 'Describing the test', example: exampleElement, source: exampleSrc }]} />);
  expect(wrapper).toMatchSnapshot();
});

it('should show multiple examples', () => {
  const wrapper = shallow(<SiteDocTemplate
    examples={[{ title: 'Test Example 1', description: 'Describing the test', example: exampleElement, source: exampleSrc },
      { title: 'Test Example 2', description: 'Describing the test mk. 2', example: exampleElement, source: exampleSrc }]}
  />);
  expect(wrapper).toMatchSnapshot();
});

it('should show one props table', () => {
  const wrapper = shallow(<SiteDocTemplate propsTables={[{ componentSource: testComponentSrc, componentName: 'Test Component' }]} />);
  expect(wrapper).toMatchSnapshot();
});

it('should show multiple props tables', () => {
  const wrapper = shallow(<SiteDocTemplate
    propsTables={[{ componentSource: testComponentSrc, componentName: 'Test Component' },
      { componentSource: testComponentSrc, componentName: 'Test Component (Again)' }]}
  />);
  expect(wrapper).toMatchSnapshot();
});

it('should show the version, readme, examples, and props tables', () => {
  const wrapper = shallow(<SiteDocTemplate
    version="1.9"
    readme={readme}
    examples={[{ title: 'Test Example 1', description: 'Describing the test', example: exampleElement, source: exampleSrc },
      { title: 'Test Example 2', description: 'Describing the test mk. 2', example: exampleElement, source: exampleSrc }]}
    propsTables={[{ componentSource: testComponentSrc, componentName: 'Test Component' },
      { componentSource: testComponentSrc, componentName: 'Test Component (Again)' }]}
  />);
  expect(wrapper).toMatchSnapshot();
});
