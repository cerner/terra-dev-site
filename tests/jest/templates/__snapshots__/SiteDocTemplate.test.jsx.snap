// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a default component with nothing 1`] = `<div />`;

exports[`should show multiple examples 1`] = `
<div>
  <h1
    id="examplesHeader"
    style={
      Object {
        "borderBottom": "1px solid #eaecef",
        "paddingBottom": "0.3em",
      }
    }
  >
    Examples
  </h1>
  <IndexExampleTemplate
    description="Describing the test"
    example={<test-file-stub />}
    exampleSrc="test-file-stub"
    key="0"
    title="Test Example 1"
  />
  <IndexExampleTemplate
    description="Describing the test mk. 2"
    example={<test-file-stub />}
    exampleSrc="test-file-stub"
    key="1"
    title="Test Example 2"
  />
</div>
`;

exports[`should show multiple props tables 1`] = `
<div>
  <PropsTable
    componentName="Test Component"
    key="0"
    src="test-file-stub"
  />
  <PropsTable
    componentName="Test Component (Again)"
    key="1"
    src="test-file-stub"
  />
</div>
`;

exports[`should show one example 1`] = `
<div>
  <h1
    id="examplesHeader"
    style={
      Object {
        "borderBottom": "1px solid #eaecef",
        "paddingBottom": "0.3em",
      }
    }
  >
    Examples
  </h1>
  <IndexExampleTemplate
    description="Describing the test"
    example={<test-file-stub />}
    exampleSrc="test-file-stub"
    key="0"
    title="Test Example 1"
  />
</div>
`;

exports[`should show one props table 1`] = `
<div>
  <PropsTable
    componentName="Test Component"
    key="0"
    src="test-file-stub"
  />
</div>
`;

exports[`should show the readme 1`] = `
<div>
  <Markdown
    src="test-file-stub"
  />
</div>
`;

exports[`should show the version 1`] = `
<div>
  <div
    id="version"
  >
    Version: 
    1.9
  </div>
</div>
`;

exports[`should show the version and readme 1`] = `
<div>
  <div
    id="version"
  >
    Version: 
    1.9
  </div>
  <Markdown
    src="test-file-stub"
  />
</div>
`;

exports[`should show the version, readme, examples, and props tables 1`] = `
<div>
  <div
    id="version"
  >
    Version: 
    1.9
  </div>
  <Markdown
    src="test-file-stub"
  />
  <h1
    id="examplesHeader"
    style={
      Object {
        "borderBottom": "1px solid #eaecef",
        "paddingBottom": "0.3em",
      }
    }
  >
    Examples
  </h1>
  <IndexExampleTemplate
    description="Describing the test"
    example={<test-file-stub />}
    exampleSrc="test-file-stub"
    key="0"
    title="Test Example 1"
  />
  <IndexExampleTemplate
    description="Describing the test mk. 2"
    example={<test-file-stub />}
    exampleSrc="test-file-stub"
    key="1"
    title="Test Example 2"
  />
  <PropsTable
    componentName="Test Component"
    key="2"
    src="test-file-stub"
  />
  <PropsTable
    componentName="Test Component (Again)"
    key="3"
    src="test-file-stub"
  />
</div>
`;
